aries/aries_vcx/src/handlers/out_of_band/receiver.rs

```diff
@@ -62,11 +84,45 @@ impl OutOfBandReceiver {
         self.oob.clone().into()
     }
 
-    pub fn from_string(oob_data: &str) -> VcxResult<Self> {
-        Ok(Self {
-            oob: serde_json::from_str(oob_data)?,
-        })
+    pub fn to_json_string(&self) -> String {
+        self.to_aries_message().to_string()
     }
+
+    pub fn to_base64_url(&self) -> String {
+        BASE64_URL_SAFE.encode(self.to_json_string())
+    }
+
+    pub fn to_url(&self, domain_path: &str) -> VcxResult<Url> {
+        let mut oob_url = Url::parse(domain_path)?;
+        let oob_query = "oob=".to_owned() + &self.to_base64_url();
+        oob_url.set_query(Some(&oob_query));
```

Comment:
nit; could you utilize the Url methods to add the query pair, rather than trying to assemble the pair yourself? e.g. `url.query_pairs_mut().append_pair("oob", self.base64)`

Reply:
Ah dang, I switched the parsing to use the query pairs, but didn't for the output to. Nice catch, that'll be a lot cleaner

Reply:
Unfortunately, it appears `query_pairs_mut()` uses the `application/x-www-form-urlencoded` syntax, which breaks when using padded base64url, as queries are strung together using `&` to separate parameters and `=` between the key and the actual value, so it throws `%3D` on the end instead of the base64url `=`. 

So I don't think we can actual do that, even though I like it better. Recommendations for how to do what I am doing now more cleanly @gmulhearn?

Reply:
IMO that sounds like how it _should_ be encoded 😅, and the Aries RFC is incorrect for giving an example with `=` padding at the end of the URL, as that seems like an invalid usage/placement of `=` in a URL string. I would think that it'd work both ways for encoding/decoding in rust though; i.e. when you do `.query_pairs().find(|(name, _value)| name == "oob")`, it would decode `%3D` back into `=` for the extracted query value string.

But i suppose other aries implementation might take the RFC example at face value and then error out when trying to parse `%3D` (this is also another reason i wish the RFC had set a requirement on _unpadded_ base64url). If you're curious, you could look at impls (credo, acapy) to see how they handle extraction of the `oob` query param to see if they would deal with the URL encoded padding `%3D`. 

End rant; if we wanna keep the `=` at the end for interop with other aries agents, then i think how you had it is fine

Reply:
I would posit that given the following from the Base64URL encoding referenced in the Aries RFC (https://datatracker.ietf.org/doc/html/rfc4648#section-5):
> The pad character "=" is typically percent-encoded when used in an
   URI [[9](https://datatracker.ietf.org/doc/html/rfc4648#ref-9)], but if the data length is known implicitly, this can be
   avoided by skipping the padding; see [section 3.2](https://datatracker.ietf.org/doc/html/rfc4648#section-3.2).

That this indicates that an agent _SHOULD_ be able to handle either no padding (which seems maybe inappropriate here given the above text and that our data length is not known implicitly due to variable invitation length) or by percent encoding, and that if they don't it is their failing, not VCX's.

Given that logic, I am going to update this PR to support encoding to with this padding and percent encoding it. I'll also probably open a PR to the Aries RFCs to adjust the example invitation (we'll see what happens there lol)

Reply:
As per Stephen Curran's guidance in https://github.com/hyperledger/aries-rfcs/pull/849, I think we're good to do unpadded as we discussed on the Community Call yesterday. :+1: